central/cluster/datastore/datastore_impl_postgres_test.go

```diff
@@ -31,31 +62,775 @@ func TestClusterDataStoreWithPostgres(t *testing.T) {
+		s.NoError(err)
+		s.NotEmpty(clusterId)
+		cluster.Id = clusterId
+		err = s.clusterDatastore.UpdateClusterHealth(ctx, clusterId, existingHealths[i])
+		s.NoError(err)
+	}
+	results, err := s.clusterDatastore.SearchRawClusters(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	for i, result := range results {
+		protoassert.Equal(s.T(), existingHealths[i], result.HealthStatus)
+	}
+}
+
+func (s *ClusterPostgresDataStoreTestSuite) TestLookupOrCreateClusterFromConfig() {
+	const bundleID = "aW5pdC1idW5kbGUtaWQK"
+	const policyVersion = "1"
+	const someHelmConfigJSON = `{
+		"staticConfig": {
+		  "type": "KUBERNETES_CLUSTER",
+		  "mainImage": "docker.io/stackrox/main",
+		  "centralApiEndpoint": "central.stackrox.svc:443",
+		  "collectorImage": "docker.io/stackrox/collector"
+		},
+		"configFingerprint": "1234"
+	  }`
+	const differentConfigFPHelmConfigJSON = `{
+		"staticConfig": {
+		  "type": "KUBERNETES_CLUSTER",
+		  "mainImage": "docker.io/stackrox/main",
+		  "centralApiEndpoint": "central.stackrox.svc:443",
+		  "collectorImage": "docker.io/stackrox/collector"
+		},
+		"configFingerprint": "12345"
+	  }`
+	var someHelmConfig storage.CompleteClusterConfig
+	var differentConfigFPHelmConfig storage.CompleteClusterConfig
+
+	err := jsonutil.JSONToProto(someHelmConfigJSON, &someHelmConfig)
+	s.NoError(err)
+
+	err = jsonutil.JSONToProto(differentConfigFPHelmConfigJSON, &differentConfigFPHelmConfig)
+	s.NoError(err)
+
+	defaultSensorId := func() *storage.SensorDeploymentIdentification {
+		return &storage.SensorDeploymentIdentification{
+			AppNamespace:       "stackrox",
+			SystemNamespaceId:  "123",
+			AppNamespaceId:     "123",
+			DefaultNamespaceId: "123",
+		}
+	}
+
+	someClusterWithManagerType := func(managerType storage.ManagerType, helmConfig *storage.CompleteClusterConfig) *storage.Cluster {
+		return &storage.Cluster{
+			Id:                 "",
+			Name:               "",
+			InitBundleId:       bundleID,
+			HelmConfig:         helmConfig,
+			MainImage:          mainImage,
+			CentralApiEndpoint: centralEndpoint,
+			ManagedBy:          managerType,
+			MostRecentSensorId: defaultSensorId(),
+			HealthStatus: &storage.ClusterHealthStatus{
+				SensorHealthStatus: storage.ClusterHealthStatus_HEALTHY,
+				LastContact:        protoconv.ConvertTimeToTimestamp(time.Now()),
+			},
+		}
+	}
+
+	sensorHelloWithHelmManagedConfigInit := func(helmManagedConfigInit *central.HelmManagedConfigInit, identification *storage.SensorDeploymentIdentification) *central.SensorHello {
+		if identification == nil {
+			identification = defaultSensorId()
+		}
+		return &central.SensorHello{
+			DeploymentIdentification: identification,
+			HelmManagedConfigInit:    helmManagedConfigInit,
+			PolicyVersion:            policyVersion,
+		}
+	}
+
+	cases := []struct {
+		description             string
+		cluster                 *storage.Cluster
+		sensorHello             *central.SensorHello
+		bundleID                string
```

Comment:
This does not seem to be used in the test body. The creation function takes the test-global variable as input, not the case-specific one.